rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,214 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+		x := int(math.Round((pt.X - minX) * scaleFactor))
+		y := int(math.Round((pt.Z - minZ) * scaleFactor))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight {
+			pointColor, err = getColorFromProbabilityValue(data)
+			if err != nil {
+				return false
+			}
+			im.Circle(image.Point{X: x, Y: y}, pointRadius, pointColor)
+		}
+		return true
+	})
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Add a red robot marker to the image
+	if ppRM.robotPose != nil {
+		x := int(math.Round((robotMarker.Point().X - minX) * scaleFactor))
+		y := int(math.Round((robotMarker.Point().Z - minZ) * scaleFactor))
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		im.Circle(image.Point{X: x, Y: y}, robotMarkerRadius, robotMarkerColor)
+	}
+	return im, nil, nil
+}
+
+// RGBDToPointCloud is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	return nil, errors.New("converting an RGB image to Pointcloud is currently unimplemented for this projection")
+}
+
+// ImagePointTo3DPoint is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
+	return r3.Vector{}, errors.New("converting an image point to a 3D point is currently unimplemented for this projection")
+}
+
+// getColorFromProbabilityValue returns an RGB color value based on the probability value and defined hit and miss
+// thresholds
+// TODO (RSDK-1705): Once probability values are available, a temporary algorithm is being used based on Cartographer's method
+// of painting images. Currently this function will return a shade of green if the probability is above the hit threshold and
+// a shade of blue if it is below the miss threshold. These shades will be more distinct the further from the threshold they are.
+func getColorFromProbabilityValue(d pointcloud.Data) (rimage.Color, error) {
+	var r, g, b uint8
+
+	if d == nil {
+		return rimage.NewColor(0, 0, 0), errors.New("data received was null")
+	}
+
+	if !d.HasValue() {
+		return rimage.NewColor(255, 255, 255), nil
+	}
+
+	if d.Value() > 100 || d.Value() < 0 {
+		return rimage.NewColor(0, 0, 0),
+			errors.Errorf("received a value of %v which is outside the range (0 - 100) representing probabilities", d.Value())
+	}
+
+	prob := float64(d.Value()) / 100.
+
+	switch {
+	case prob < missThreshold:
+		b = uint8(255 * ((missThreshold - prob) / (hitThreshold - 0)))
+	case prob > hitThreshold:
+		g = uint8(255 * ((prob - hitThreshold) / (1 - missThreshold)))
+	default:
+		b = uint8(255 * ((missThreshold - prob) / (hitThreshold - 0)))
+		g = uint8(255 * ((prob - hitThreshold) / (1 - missThreshold)))
```

Comment:
Won't `prob - hitThreshold` result in a negative number here? Won't that result in a panic?